swyft/utils/utils.py

```diff
@@ -157,5 +158,24 @@ def estimate_coverage(
     return cont_fraction
 
 
+def estimate_empirical_mass(dataset, post, nobs, npost):
+    obs0, u0, v0 = dataset[0]
+    w0 = post.eval(v0.unsqueeze(0).numpy(), obs0)["weights"]
+    mass = {
+        k: dict(nominal=[], empirical=np.linspace(1 / nobs, 1, nobs)) for k in w0.keys()
+    }
+    for i in range(nobs):
+        j = np.random.randint(len(dataset))
+        obs0, u0, v0 = dataset[j]
+        w0 = post.eval(v0.unsqueeze(0).numpy(), obs0)["weights"]
+        wS = post.sample(npost, obs0)["weights"]
+        for k, v in w0.items():
+            f = wS[k][wS[k] >= v].sum() / wS[k].sum()
+            mass[k]["nominal"].append(f)
+    for k in mass.keys():
+        mass[k]["nominal"] = sorted(mass[k]["nominal"])
```

Comment:
this returns a list. I think we should have both be numpy arrays. I'll make the change.

Reply:
changed